pandas/io/json/_json.py

```diff
@@ -801,9 +814,48 @@ def __init__(
             self.nrows = validate_integer("nrows", self.nrows, 0)
             if not self.lines:
                 raise ValueError("nrows can only be passed if lines=True")
+        if self.engine == "pyarrow":
+            if not self.lines:
+                raise ValueError(
+                    "currently pyarrow engine only supports "
+                    "the line-delimited JSON format"
+                )
+        if self.engine not in ["pyarrow", "ujson"]:
```

Comment:
very minor nit but unless order matters would rather we check against a set not a list

Reply:
I will change it to a set, I don't think the order matters. Just for learning purposes, is there a reason to compare it to a set? Because of time complexity?

Reply:
Yeah agreed here. Could you use a `set`?